webapi/Plugins/Utils/JsonUtils.cs

```diff
@@ -0,0 +1,113 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Text.Json;
+using System.Text.RegularExpressions;
+
+namespace CopilotChat.WebApi.Plugins.Utils;
+
+/// <summary>
+/// Utility methods for working with asynchronous operations and callbacks.
+/// </summary>
+public static class JsonUtils
+{
+    /// <summary>
+    /// Try to optimize json from the planner response
+    /// based on token limit
+    /// </summary>
+    internal static string OptimizeOdataResponseJson(string jsonString, int tokenLimit)
+    {
+        // Remove all new line characters + leading and trailing white space
+        jsonString = Regex.Replace(jsonString.Trim(), @"[\n\r]", string.Empty);
+        var document = JsonDocument.Parse(jsonString);
+        bool trimResponse = false;
+
+        // The json will be deserialized based on the response type of the particular operation that was last invoked by the planner
+        // The response type can be a custom trimmed down json structure, which is useful in staying within the token limit
+        Type responseType = typeof(object);
+
+        if (trimResponse)
```

Comment:
It looks like `trimResponse` is always false, which means whatever is in the if condition is dead code.